File: csharp/Slimmed/PublishingHouse.Infrastructure/Events/EventEnvelope.cs
using PublishingHouse.Core.ValueObjects;

namespace PublishingHouse.Core.Events;

public interface IEventEnvelope
{
    object Event { get; }
    EventMetadata Metadata { get; }
}

public record EventEnvelope<TEvent>(TEvent Event, EventMetadata Metadata): IEventEnvelope
    where TEvent: notnull
{
    object IEventEnvelope.Event => Event;
}

public record EventMetadata(NonEmptyGuid RecordId);

File: java/src/main/java/io/eventdriven/slimdownaggregates/original/infrastructure/events/EventEnvelope.java
package io.eventdriven.slimdownaggregates.original.infrastructure.events;

import io.eventdriven.slimdownaggregates.original.infrastructure.valueobjects.NonEmptyString;

public interface EventEnvelope {
  Object event();

  EventMetadata metadata();


  record EventMetadata(NonEmptyString recordId) {
  }
}

